src/Common/API/CoreAPI.ts

```diff
@@ -138,10 +184,13 @@ class CoreAPI {
         isMultipartRequest,
     }: FetchInTimeParamsType<T>): Promise<ResponseType> => {
         const controller = options?.abortControllerRef?.current ?? new AbortController()
-        const signal = options?.abortControllerRef?.current?.signal || options?.signal || controller.signal
-        const timeoutPromise: Promise<ResponseType> = new Promise((resolve, reject) => {
-            const timeout = options?.timeout || this.timeout
+        const timeoutSignal = controller.signal
```

Comment:
We can remove this creation here as we are anyways creating a new controller in `mergeAbortSignals`